nuxeo-webdav/src/main/java/org/nuxeo/ecm/webdav/resource/ExistingResource.java

```diff
@@ -189,16 +238,21 @@ private Response copyOrMove(String method, @HeaderParam("Destination") String de
             return Response.status(409).build();
         }
 
-        if ("COPY".equals(method)) {
-            DocumentModel destDoc = backend.copyItem(doc, destParentRef);
-            backend.renameItem(destDoc, getNameFromPath(destPath));
-        } else if ("MOVE".equals(method)) {
-            if (backend.isRename(doc.getPathAsString(), destPath)) {
-                backend.renameItem(doc, getNameFromPath(destPath));
-            } else {
-                backend.moveItem(doc, destParentRef);
-            }
+        switch (method) {
+            case "COPY":
+                backend.copyItem(doc, destParentRef);
+                break;
+            case "MOVE":
+                String docPath = doc.getPathAsString();
+                DocumentModel newDoc = backend.moveItem(doc, destParentRef, getNameFromPath(destPath));
+                if (destPath.endsWith(".tmp")) {
+                    newDoc.setPropertyValue("dc:source", docPath);
```

Comment:
Extract the get/set of dc:source into separate methods as they are implementation details that we could want to change without digging into the rest of the code.
Also don't store the full doc path, just the last path component. And when doing `new PathRef(dcSource.toString())` resolve the name from the parent path (and check the name doesn't contain a /).